### PR TITLE
[8.x.x] [VFX] Add Update Mode inspector in VisualEffectAsset

### PR DESCRIPTION
**Pull request by PaulDemeulenaere**
### Purpose of this PR
Fix inconsistent UX for Visual Effect Asset.

*Before*
8old

*After*
image

It's only the update inspector for VisualEffectAsset, in order to fix this issue : https://fogbugz.unity3d.com/f/cases/1258463/

This change has been isolated from this PR : https://github.cds.internal.unity3d.com/unity/vfx-graphics/pull/4 already landed in branch 9x.x & master (10.x.x)

---
### Testing status

**Manual Tests**: What did you do?
- [x] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [x] Tested UI multi-edition + Undo/Redo + Prefab overrides + Alignment in Preset
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in developer mode, you have a button at end of resources that check the paths)

### CODE REVIEW

**Comment by PaulDemeulenaere** [comment]
It can be merged *after* 8.2.0, it isn't really a blocking issue.

**Comment by VladNeykov** [comment]
image
If **Exact Fixed Time** is only relevant when **Fixed Delta Time** is enabled, it should probably be hidden or indented and grayed out when Fixed Delta Time is disabled.

**Comment by VladNeykov** [comment]
Looks good otherwise, no issues found (exact fixed time is hard to test though).

We discussed also the possibility of not hiding Exact Fixed Time, but graying it out if Fixed Delta Time is not on:
image
(not a big deal, there isn't a Unity-wide consensus, but it aids discoverability if the property is not hidden, but grayed out).

File: com.unity.visualeffectgraph/Editor/Inspector/VFXAssetEditor.cs [hunk 1]
```diff
@@ -414,13 +414,102 @@ public override void OnInspectorGUI()
...
+        {
+            initialUpdateMode = (VFXUpdateMode)resourceUpdateModeProperty.intValue;
+            initialFixedDeltaTime = !((initialUpdateMode & VFXUpdateMode.DeltaTime) == VFXUpdateMode.DeltaTime);
+            initialProcessEveryFrame = (initialUpdateMode & VFXUpdateMode.ExactFixedTimeStep) == VFXUpdateMode.ExactFixedTimeStep;
+            initialIgnoreGameTimeScale = (initialUpdateMode & VFXUpdateMode.IgnoreTimeScale) == VFXUpdateMode.IgnoreTimeScale;
+        }
+
+        EditorGUI.showMixedValue = !initialFixedDeltaTime.HasValue;
+        var deltaTimeContent = EditorGUIUtility.TrTextContent("Fixed Delta Time", "If enabled, use visual effect manager fixed delta time mode.");
+        var processEveryFrameContent = EditorGUIUtility.TrTextContent("Exact Fixed Time", "Only relevant for fixed delta time mode, if enabled, can process several updates per frame (e.g.: if a frame is 30ms but the fixed frame rate is set to 5ms, it will update 6 times with a 5ms deltaTime instead of one time with a 30ms deltaTime). This update mode isn't efficient and should only be used for really high quality purpose.");
```
[the remark this answers is not in the document]
Comment (PaulDemeulenaere) [reply com.unity.visualeffectgraph/Editor/Inspector/VFXAssetEditor.cs:450]: Oops, we did well to read this twice, I will fix in this PR and update in vfx/staging as well.

**Comment by PaulDemeulenaere** [comment]
I think I should follow this recommandation even it already landed in more recent package : https://github.com/Unity-Technologies/Graphics/pull/1131#issuecomment-662607733
I will create another PR for additional changes in 10.x.x and fix directly on this one for 8.x.x

**Pull request closed without merging**